e2e/cluster_test.go

```diff
@@ -412,6 +413,110 @@ var _ = Describe("SpiceDBClusters", func() {
 		})
 	})
 
+	Describe("With missing secret", func() {
+		testNamespace := "test"
+		rv := "0"
+
+		BeforeEach(func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			DeferCleanup(cancel)
+			aec := &v1alpha1.SpiceDBCluster{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       v1alpha1.SpiceDBClusterKind,
+					APIVersion: v1alpha1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: testNamespace,
+				},
+				Spec: v1alpha1.ClusterSpec{
+					Config:    []byte(`{"datastoreEngine": "memory"}`),
+					SecretRef: "nonexistent",
+				},
+			}
+			u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(aec)
+			Expect(err).To(Succeed())
+			_, err = client.Resource(v1alpha1ClusterGVR).Namespace(testNamespace).Create(ctx, &unstructured.Unstructured{Object: u}, metav1.CreateOptions{})
+			Expect(err).To(Succeed())
+			DeferCleanup(client.Resource(v1alpha1ClusterGVR).Namespace(testNamespace).Delete, ctx, "test", metav1.DeleteOptions{})
+		})
+
+		It("Reports missing secret on the status", func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+			DeferCleanup(cancel)
+
+			var c *v1alpha1.SpiceDBCluster
+			watcher, err := client.Resource(v1alpha1ClusterGVR).Namespace(testNamespace).Watch(ctx, metav1.ListOptions{
+				Watch:           true,
+				ResourceVersion: rv,
+			})
+			Expect(err).To(Succeed())
+			foundMissingSecret := false
+			for event := range watcher.ResultChan() {
+				Expect(runtime.DefaultUnstructuredConverter.FromUnstructured(event.Object.(*unstructured.Unstructured).Object, &c)).To(Succeed())
+				if c.Name != "test" {
+					continue
+				}
+				GinkgoWriter.Println(c)
```

Comment:
Are the various calls to ginkowriter here for debugging? Not sure if you intended to remove them

Reply:
It's probably excessive, but I have the prints so that you can see the state being checked by the test in the test logs (in case it fails)